stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1346,6 +1364,7 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
     /// - parameter data: The replacement data.
     @inline(__always)
     public mutating func replaceSubrange(_ subrange: Range<Index>, with data: Data) {
+        if !(subrange.lowerBound == endIndex && subrange.count == 0) { _validateRange(subrange) }
```

Comment:
What's special about the zero-length range at the end of the bounds, as opposed to any other zero-length range? What's different between the methods that allow this and the methods that don't?